chore: upgrade gnosis dependencies and move to peerDependencies

MeLlamoPablo opened the pull request:
### Description

This PR upgrades `@safe-global/safe-apps-sdk` (f.k.a. `@gnosis.pm/safe-apps-sdk`) to `8.0.0` and `@safe-global/safe-apps-provider` (f.k.a. `@gnosis.pm/safe-apps-provider`) to `0.17.1`.

It also moves them from `dependencies` to `peerDependencies` as this Onboard module shouldn't dictate the Safe Apps SDK version a dApp uses. A dApp may want to load the Safe Apps SDK independently of Onboard and thus listing it on `dependencies` would result in either two different versions being loaded or the dApp being unable to upgrade the Onboard version.

Note that this is a breaking change as the consumers of this package must install `@safe-global/safe-apps-sdk` and `@safe-global/safe-apps-provider` independently.

#### **_PLEASE NOTE- Checklist must be complete prior to review._**
### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
     <img width="1512" alt="Screenshot 2023-06-23 at 13 38 04" src="https://github.com/blocknative/web3-onboard/assets/11708035/4883158c-4e05-4a2d-8eee-b3d00328b9ed">
- [x] Add/update the appropriate package README (if applicable) - Not applicable
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable) - Not applicable
- [x] Add/update the related package in the `docs/package.json` file (if applicable) - Not applicable

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### Tests with demo app (SDK)

**Kindly requesting help with this, as I can't get the dev server running without errors**

- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

taylorjdawson commented:
@MeLlamoPablo Awesome! We'll test the sdk part on our end and if it looks good we'll merge, thanks for the work!

Adamj1232 commented on packages/gnosis/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/gnosis",
-  "version": "2.1.10",
+  "version": "3.0.0-alpha.0",
```
```suggestion
  "version": "2.2.0-alpha.1",
```

Adamj1232 commented:
@MeLlamoPablo are there any build env settings with the latest packages that need to be noted [here](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/routes/docs/%5B...3%5Dmodules/%5B...1%5Dcore/%2Bpage.md#build-environments) in the documentation with the new deps?

MeLlamoPablo commented:
Hey guys, I'm currently OOO, just letting you know that I'll take a look as soon as I can, thanks for your comments!

MeLlamoPablo commented:
> @MeLlamoPablo are there any build env settings with the latest packages that need to be noted [here](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/routes/docs/%5B...3%5Dmodules/%5B...1%5Dcore/%2Bpage.md#build-environments) in the documentation with the new deps?

I don't think so. Looking through the source code for `@safe-global/safe-apps-provider` and `@safe-global/safe-apps-sdk` I don't see any reliance on packages not explicitly listed in `dependencies`, plus their docs don't mention any special installation instructions ([sdk](https://github.com/safe-global/safe-apps-sdk/tree/main/packages/safe-apps-sdk#install), [provider](https://github.com/safe-global/safe-apps-sdk/tree/main/packages/safe-apps-provider#how-to-use)).

MeLlamoPablo commented:
> Hey, this is Yago from Safe. That's okay with us. Would it be difficult for you to update all the "Gnosis" references to "Safe", including the package name? As you probably know, we are now a separate DAO ☺️
> 
> You can remove as well the `gnosis-safe.io` whitelisted domain as it does not exist anymore

Removed references to the domain on 4a11e18 and updated docs and examples on 3e1b9bf. Renaming the package itself is something that the Blocknative team should do.

taylorjdawson commented:
@MeLlamoPablo Is this ready for another review? thanks!

MeLlamoPablo commented:
Yeah go ahead!

El 21 de julio de 2023 21:49:33 CEST, Taylor Dawson ***@***.***> escribió:
>@MeLlamoPablo Is this ready for another review? thanks!
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/blocknative/web3-onboard/pull/1785#issuecomment-1646170051
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

yagopv commented:
Hey @MeLlamoPablo thanks for the renaming !! ... really appreciate

Adamj1232 commented:
@MeLlamoPablo so you want the name of the package to be changed to Safe? 
I feel like just changing the likes and implementations is a little confusing.
If so we will need to mark the package as deprecated and create(copy) a new `Safe` module.
This will require additional overhead and testing but is probably worth it

MeLlamoPablo commented:
> @MeLlamoPablo so you want the name of the package to be changed to Safe? I feel like just changing the likes and implementations is a little confusing. If so we will need to mark the package as deprecated and create(copy) a new `Safe` module. This will require additional overhead and testing but is probably worth it

I think the package should eventually be renamed to `@web3-onboard/safe`, but that should be done on another PR.

yagopv commented:
Opened an issue for you regarding the renaming https://github.com/blocknative/web3-onboard/issues/1849. Thanks!

taylorjdawson commented:
@MeLlamoPablo Once the merge conflicts are resolved this looks good to go 🙌

MeLlamoPablo commented:
> @MeLlamoPablo Once the merge conflicts are resolved this looks good to go 🙌

Solved!
